src/Writer/PhpArray.php

```diff
@@ -139,4 +158,54 @@ protected function processIndented(array $config, array $arraySyntax, &$indentLe
 
         return $arrayString;
     }
+
+    /**
+     * Process a string configuration value
+     *
+     * @param string $value
+     * @return string
+     */
+    protected function processStringValue($value)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($value)) {
+            return $value . '::class';
+        }
+
+        return var_export($value, true);
+    }
+
+    /**
+     * Process a string configuration key
+     *
+     * @param string $key
+     * @return string
+     */
+    protected function processStringKey($key)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($key)) {
+            return $key . '::class';
+        }
+
+        return "'" . addslashes($key) . "'";
+    }
+
+    /**
+     * Check whether a string represents a resolvable FQCN
+     *
+     * @param string $string
+     * @return bool
+     */
+    protected function checkStringIsFqn($string)
```

Comment:
You really need to match against the regex 
[`'/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]+$/'`](https://github.com/Ocramius/ProxyManager/blob/842a41b73713940352c9589a6e3805fb0c69a29a/tests/ProxyManagerTest/Generator/Util/UniqueIdentifierGeneratorTest.php#L59) ([explained](http://regexper.com/#%2F^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]%2B%24%2F))

Grabbed it from http://stackoverflow.com/questions/17973357/what-are-the-valid-characters-in-php-variable-method-class-etc-names, can't remember if it's also in the docs.